src/applications/vaos/appointment-list/redux/selectors.js

```diff
@@ -385,47 +383,93 @@ export function selectAppointmentLocality(appointment) {
   const isVideo = selectIsVideo(appointment);
   const isInPerson = selectIsInPerson(appointment);
 
-  if (isCommunityCare) return 'Community care';
-  if (isPhone) return 'VA Appointment';
-  if (isVideo)
-    return practitioner
-      ? `VA Appointment with ${practitioner}`
-      : 'VA Appointment';
-  if (isInPerson)
-    return typeOfCareName && practitioner
-      ? `${typeOfCareName} with ${practitioner}`
-      : 'VA Appointment';
+  if (
+    typeOfCareName &&
+    practitioner &&
+    (isInPerson || isVideo || isPhone || isCommunityCare)
+  ) {
+    return `${typeOfCareName} with ${practitioner}`;
```

Comment:
To get the last name of the provider would be something like below.
```suggestion
    return `${typeOfCareName} with ${practitioners[0].name.family}`;
```